lib/puppet/functions/conjur/client.rb

```diff
@@ -2,16 +2,23 @@
 
 require 'net/http'
 
-# This function is really a Ruby class. Since we can't bind a class
-# to a Ruby constant (which would dirty the interpreter state),
-# we make it anonymous module and 'bless' a hash with it.
+# This function is a representation of a Conjur / DAP client with majority
+# of the functionality that allows us to authenticate against a server.
 Puppet::Functions.create_function :'conjur::client' do
+  # @param uri The URL of the Conjur or DAP instance.
+  # @param version Conjur API version.
+  # @param cert The _raw_ PEM-encoded x509 CA certificate chain for the DAP instance
+  # @return [Class] Conjur client instance
   dispatch :new do
     param 'String', :uri
     param 'Integer', :version
     param 'String', :cert
   end
 
+  # @param uri The URL of the Conjur or DAP instance.
+  # @param version Conjur API version.
+  # @param cert The _raw_ PEM-encoded x509 CA certificate chain for the DAP instance
```

Comment:
Does this comment need to indicate that `cert` parameter is optional?

Reply:
I'm not 100% sure that new logic can allow for a missing SSL cert so for now it's just not documented that you can do that. This will be something to test out as part of the XA.